server/tests/api_v4/authenticated/test_user_revoke.py

```diff
@@ -75,3 +75,37 @@ async def test_disconnect_sse(
 
     rep = await coolorg.bob.raw_sse_connection()
     assert rep.status_code == 461
+
+
+async def test_authenticated_user_revoke_user_already_revoked(
+    coolorg: CoolorgRpcClients, backend: Backend
+) -> None:
+    t1 = DateTime.now()
+    certif1 = RevokedUserCertificate(
+        author=coolorg.alice.device_id,
+        timestamp=t1,
+        user_id=coolorg.bob.device_id.user_id,
+    )
+
+    outcome = await backend.user.revoke_user(
```

Comment:
This test is testing the rpc client, isn't it ? so why not using `coolorg.alice.user_revoke` twice?

Reply:
It felt weird at first... it seemed to me that is more interesting to test the (lack of) _idempotence_ of the `user_revoke` (rpc) call rather than modifying state + testing that the function behaves as expected. Calling `coolorg.alice.user_revoke` twice would also tests that the function has set the correct state (revoke certificate) in the first call. 

I guess you could say that this is already tested in the "ok" test. 
So, in the end both seem more or less equivalent :+1: